indexer/processors/bridge.go

```diff
@@ -0,0 +1,119 @@
+package processors
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/indexer/database"
+	"github.com/ethereum-optimism/optimism/indexer/processors/bridge"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type BridgeProcessor struct {
+	log log.Logger
+	db  *database.DB
+
+	// reorgs?
+	// On each loop we check that the recorded hash exists? If not,
+	// we assume a reorg and start from the latest known headers
+
+	LastEpoch *database.Epoch
+}
+
+func NewBridgeProcessor(log log.Logger, db *database.DB) (*BridgeProcessor, error) {
+	log = log.New("processor", "bridge")
+
+	// TODO: Just start at genesis. This processor wont be restartable until
+	// we can fill in the starting point
+	return &BridgeProcessor{log: log, db: db, LastEpoch: nil}, nil
+}
+
+func (b *BridgeProcessor) Start(ctx context.Context) error {
```

Comment:
One thing that I think is missing from the processor in general is the idea of "catching up" while also processing at head. Ideally when you run the indexer it both is always indexing new events right away while also catching up from it's latest fully synced state. Not only does this make the indexer 'good enough' for users right away even in a full resync in many cases but it also makes the UI more testable e2e tests vs live networks easier and hardens the code by guaranteeing it always gracefully handles processing blocks out of order

Reply:
ooh yea totally this is a good point and requires some design. Let's file an issue for this as a good followup / nice to have after launch

Kinda like the my reply about speeding up the ETL, I can imagine you start indexing at HEAD then parallelize back-filling everything in the past split across a set of workers